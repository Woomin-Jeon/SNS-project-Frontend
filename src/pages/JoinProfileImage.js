import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import func from '../function';

function JoinProfileImage({
  currentUserState,
  setCurrentUserState,
}) {
  const [file, setFile] = useState('');
  const [uploadedFile, setUploadedFile] = useState({});
  const history = useHistory();

  const { id } = currentUserState;

  const onChange = async (e) => {
    await setFile(e.target.files[0]);
    const send = document.getElementById('send');
    send.click();
  };

  const onSubmit = async (e) => {
    e.preventDefault();

    const formData = new FormData();
    formData.append('file', file);
    const { fileName, filePath } = await func.fileUpload(formData);
    await func.addProfileImage(id, filePath);

    setCurrentUserState({
      ...currentUserState,
      profileImage: filePath,
    });
    setUploadedFile({ fileName, filePath });
  };

  const moveNextPage = () => {
    history.push('/joinfollow');
  };


  return (
    <>
      <form onSubmit={onSubmit}>
        <div>
          <input type="file" onChange={onChange} />
          <input className="hidden" value="" id="send" type="submit" />
        </div>
        {uploadedFile ? (
          <div>
            <div>
              <img style={{ width: '100%' }} src={uploadedFile.filePath} alt="" />
            </div>
          </div>
        ) : null}
      </form>
      <button onClick={moveNextPage}>
        다음
      </button>
    </>
  );
}

export default JoinProfileImage;
