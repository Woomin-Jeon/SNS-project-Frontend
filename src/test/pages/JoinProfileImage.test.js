import JoinProfileImage from "../../pages/JoinProfileImage";
import React from "react";
import {shallow} from 'enzyme';
import {BrowserRouter} from "react-router-dom";

describe('JoinProfileImage', () => {
  let currentUserState;
  let setCurrentUserState;

  it('renders well', () => {
    const component = shallow(
      <BrowserRouter>
        <JoinProfileImage
          currentUserState={currentUserState}
          setCurrentUserState={setCurrentUserState}
        />
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
