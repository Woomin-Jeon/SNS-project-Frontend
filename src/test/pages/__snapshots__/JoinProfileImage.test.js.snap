// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinProfileImage renders well 1`] = `ShallowWrapper {}`;
